### PR TITLE
User from user's agent, and expanding third parties

### PR DESCRIPTION
**Pull request by jwrosewell**
Started to separate out "users" from "user's agent".

Expanding third party to consider all parties that are not first party origins.

***
<a href="https://pr-preview.s3.amazonaws.com/jwrosewell/security-questionnaire/pull/94.html" title="Last updated on Feb 2, 2021, 12:34 AM UTC (7192d64)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/security-questionnaire/94/3b4e968...jwrosewell:7192d64.html" title="Last updated on Feb 2, 2021, 12:34 AM UTC (7192d64)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -286,16 +380,16 @@ serve as an identifier if misused/abused [[OLEJNIK-BATTERY]].
 </p>
 
 <h3 class=question id="other-data">
-  What data does this specification expose to an origin?  Please also
-  document what data is identical to data exposed by other features, in the
-  same or different contexts.
+  What data does this specification expose to an origin or other party?  
+  Please also at data is identical to data exposed by other features, in
```
Comment (brodrigu) [inline index.bs:384]: typo here?

**Comment by joshuakoran** [comment]
I think there are two good points being clarified in the edits:

- Distinguishing identification of a particular user from identification of a particular person's user agent across different sessions or interactions with different origins (aka publisher domains) 
- Given end user's perspective, ensuring fingerprinting is addressed regardless of whether it is attempted across different origins or not   

I think the distinctions raised in these suggested edits are worth clarifying in the document.

File: index.bs [hunk 2]
```diff
@@ -136,77 +137,170 @@ important to consider ways to mitigate the obvious impacts. For instance:
...
+trans status
+are all examples of sensitive information.
+
+When a feature exposes sensitive information to the web,
+its designers must take steps
+to mitigate the risk of exposing the information.
+
+<div class=example>
+
+The Credential Management API allows sites
```
Comment (jyasskin) [inline index.bs:182]: This PR seems to mix stylistic rewordings like this paragraph, with changes in meaning like "or other party" in the first paragraph and "user"->"user agent" in the fingerprinting section. That makes it hard to review.

Comment (jwrosewell) [reply index.bs:182]: Perhaps we could arrange time, either in a side meeting, or at a future meeting to discuss this? Personally I find document collaboration via GitHub to be more complex than other document collaboration tools such as those provided by Google or Microsoft.

In the meantime here is some background and related issues.

**PR Background** 

The PR is made against the "[revise-2](https://github.com/w3ctag/security-questionnaire/tree/revise-2)" branch of the master document. This PR introduces the following for reviewer consideration in addition to the changes already included in "revise-2".

1. Differentiate between "user" and "user agent". For example; a fingerprint can relate to a person, or a user agent. The harms, benefits and risks we are concerned about vary accordingly.

2. There are three different entities that might benefit or harm people’s privacy.

a)	First-parties – web site authors who are readily identifiable via the domain name displayed in the address bar.
b)	First-parties’ suppliers – organisations first-parties rely upon that are not identified in the address bar. 
c)	User agent vendors – organisations that make or control the web browser

People can be harmed by any one or all three of the above. Afterall some user agent vendors have received the largest fines in relation to privacy violations. 

The document would benefit from recognising each group and the role they play in the specific issues of security and privacy. This was originally raised as an [issue](https://github.com/w3ctag/security-questionnaire/issues/84) which includes some background.

**Related Issues**

These related issues are important. @torgo closed an issue related to [people's ability to trust supply chains](https://github.com/w3ctag/security-questionnaire/issues/83) with this statement “We have already established that the statement "supply chains can be trusted" (on its own) is false.” But unfortunately, did not provide me with any references to justify that that statement is false. Such information should be included in the document and be provided from an authoritative source. We can look at other industries they all have supply chains people can trust. Why should the web be different? As a minimum the document should be expanded to consider the conditions that do or do not enable a supply chain to be trust.

Referenced [RFC 6973](https://tools.ietf.org/html/rfc6973) entertains the possibility and certainly does not exclude it.

I believe the onus in a consensus driven governance model is for the proposer to convince others. As a new member I am keen to understand the facts and feel these issues have been closed without discussion or clarity in the document. I've been advised to progress resolution via text changes to the document and I'm doing so here.

**Policy**

The more general issue relates to W3C policy. This document infers policy. It does not seem to be the role of a standards body to make decisions that restrict people’s choice. That is a role of law makers.

Whilst these documents are supposed to advise they do prescribe rules akin to laws. According to the [W3C Process](https://w3c.github.io/w3process/index.html) if a proposal is to be successful it will at some point need to pass horizontal review. It's unlikely the AC or the Director would approve a standard that were deficient.

Rightly proposers become familiar with those documents that will be used to assess their proposal at the onset. Reviewer's frequently reference these documents. Whilst these documents may not be formally classified as laws, they appear to be use as such in practice.

File: index.bs [hunk 3]
```diff
@@ -314,7 +315,7 @@ communication methods.
 
 When a specification exposes specific information about a host to an origin,
 if that information changes rarely and is not variable across origins, then
-it can be used to uniquely identify a user across two origins — either
+it can be used to uniquely identify a user's agent across two origins — either
```
Comment (torgo) [inline index.bs:318]: Firstly it should specifically be talking about user-agent (vs user's agent) as that is a term of art. Secondly, in some cases we could be talking about users and in some cases user agents - as for example when a unqiue identifier could be accessible to two user agents.

File: index.bs [hunk 4]
```diff
@@ -91,8 +91,9 @@ we've prepared [a list of these questions in Markdown](https://raw.githubusercon
 </h3>
 
 Just because information can be exposed to the web doesn’t mean that it
-should be. How does exposing this information to an origin benefit a user?
-Is the benefit outweighed by the potential risks?  If so, how?
+should be. How does exposing this information to an origin or other party
```
Comment (hober) [inline index.bs:94]: Could you rebase? I don't think this change is necessary given the current text.

File: index.bs [hunk 5]
```diff
@@ -332,7 +333,7 @@ exfiltrate data.
 <p class=example>
 The `RENDERER` string exposed by some WebGL implementations
 improves performance in some kinds of applications, but does so at the
-cost of adding persistent state to a user's fingerprint. These kinds of
+cost of adding persistent state to a user agent's fingerprint. These kinds of
```
Comment (torgo) [inline index.bs:336]: This looks fine.

File: index.bs [hunk 6]
```diff
@@ -351,11 +352,11 @@ entropy introduced by [disallowing direct enumeration of the plugin list](https:
...
-Information from sensors may serve as a fingerprinting vector across origins.
-In addition, sensor also reveals something about my device or environment and
-that fact might be what is sensitive.  In addition, as technology advances,
-mitigations in place at the time a specification is written may have to be
-reconsidered as the threat landscape changes.
+Information from sensors may serve as a fingerprinting vector on the same origin
+or across origins.  In addition, sensor also reveals something about the user’s 
+agent or environment and that fact might be what is sensitive.  In addition, as 
+technology advances, mitigations in place at the time a specification is written 
+may have to be reconsidered as the threat landscape changes.
```
Comment (torgo) [inline index.bs:359]: This is fine with a few tweaks - e.g. user's agent -> user agent

File: index.bs [hunk 7]
```diff
@@ -351,11 +352,11 @@ entropy introduced by [disallowing direct enumeration of the plugin list](https:
 If so, what kind of sensors and information derived from those sensors does
 this standard expose to origins?
 
-Information from sensors may serve as a fingerprinting vector across origins.
-In addition, sensor also reveals something about my device or environment and
-that fact might be what is sensitive.  In addition, as technology advances,
-mitigations in place at the time a specification is written may have to be
-reconsidered as the threat landscape changes.
+Information from sensors may serve as a fingerprinting vector on the same origin
+or across origins.  In addition, sensor also reveals something about the user’s
```
Comment (hober) [inline index.bs:356]: Maybe s/on the same origin or across origins/on origins/, like in #90?

Comment (hober) [reply index.bs:356]: Also, there's some grammar weirdness in the original text too ("sensor also reveals" should probably be "sensors can reveal". Can you fix this while you're in here? Thanks!

File: index.bs [hunk 8]
```diff
@@ -379,16 +380,16 @@ serve as an identifier if misused/abused [[OLEJNIK-BATTERY]].
 </p>
 
 <h3 class=question id="other-data">
-  What data does this specification expose to an origin?  Please also
-  document what data is identical to data exposed by other features, in the
-  same or different contexts.
+  What data does this specification expose to an origin or other party?
```
Comment (torgo) [inline index.bs]: I think "origin" is fine since "parties" are origins in the web.

File: index.bs [hunk 9]
```diff
@@ -379,16 +380,16 @@ serve as an identifier if misused/abused [[OLEJNIK-BATTERY]].
...
 </h3>
 
-As noted above in [[#sop-violations]], the [=same-origin policy=] is an
+As noted in [[#sop-violations]], the [=same-origin policy=] is an
 important security barrier that new features need to carefully consider.
 If a specification exposes details about another origin's state, or allows
-POST or GET requests to be made to another origin, the consequences can be
-severe.
+POST or GET requests to be made to another origin or party, the consequences
+can be severe.
```
Comment (torgo) [inline index.bs:392]: this is fine.

File: index.bs [hunk 10]
```diff
@@ -379,16 +380,16 @@ serve as an identifier if misused/abused [[OLEJNIK-BATTERY]].
...
-  What data does this specification expose to an origin?  Please also
-  document what data is identical to data exposed by other features, in the
-  same or different contexts.
+  What data does this specification expose to an origin or other party?  
+  Please also at data is identical to data exposed by other features, in 
+  the same or different contexts.
 </h3>
 
-As noted above in [[#sop-violations]], the [=same-origin policy=] is an
+As noted in [[#sop-violations]], the [=same-origin policy=] is an
```
Comment (torgo) [inline index.bs:388]: this is fine

File: index.bs (hunk 9 above)
Comment (torgo) [inline index.bs:392]: Also not necessary as origin implies party

File: index.bs [hunk 11]
```diff
@@ -501,21 +502,18 @@ at zero, increments, and is reset — is a good example of a privacy friendly
 temporary identifier.
 
 <h3 class=question id="first-third-party">
-  How does this specification distinguish between behavior in first-party and
-  third-party contexts?
+  How does this specification protect end users from storing persistent data 
+  on a user’s device across browsing session?
```
Comment (hober) [inline index.bs:506]: While I think your new question may be one worth asking in the questionnaire, I'd rather it be spun off into a separate PR that *adds* this question, instead of one which *replaces* the first-/third-party question. Among other things, this new question is significantly narrower than the original.

Comment (hober) [reply index.bs:506]: Maybe pursue removing or modifying the first-/third-party question in a separate PR as well.

File: index.bs (hunk 3 above)
Comment (jwrosewell) [reply index.bs]: Resolved in latest commit.

File: index.bs (hunk 6 above)
Comment (jwrosewell) [reply index.bs:359]: Resolved in latest commit.

File: index.bs (hunk 7 above)
Comment (jwrosewell) [reply index.bs]: Fixed "sensors can reveal" comment in latest commit.

File: index.bs (hunk 8 above)
Comment (jwrosewell) [reply index.bs]: Other parties include the user agent vendor that I don't believe would be classified as origins. Consider a feature where the user agent vendor is capturing personal information. Considerations related to privacy would at least include;

a) consent obtained for use of that information;
b) transparency to assure the user that the consent preferences were respected; and
c) security issues associated with the way that data is stored and deleted, among others. 

Considering security and privacy beyond protocols and origins is an important theme I would like to see developed in PING and this document. What mitigations can be applied that do not relate entirely to technical standards? For example; accepting terms of service and default settings at the point of user agent setup or upgrade will enable implementors to consider solutions that are not restricted to permission pop ups that are annoying, or defaults that are so restrictive they impair functionality, or expose user agent vendors to competition issues by gaining consent across vertically integrated services at the point of account setup which is impossible for other players in the market to achieve.

I hope we can all agree that a situation where acceptable privacy can only be achieved by a very small number of implementors would not be compatible with the W3C antitrust policy or the mission and purpose of the W3C.

File: index.bs (hunk 11 above)
Comment (jwrosewell) [reply index.bs:507]: There are two key changes in this pull request.

1.	Recognising the full range of parties involved in the web.
2.	Resolving first and third-party bias.

Before splitting the second change into another pull request I’m keen to gauge wider views on this and have been encouraged to raise these issues via specific text changes. 

If smaller participants on the web are denied the ability to “band together” to deliver services that only larger participants can deliver due to their scale, then we have to consider the W3C’s antitrust policy. What work has been done on this document to verify it does not unconsciously steer specifications towards this outcome? Whilst I recognise that the subject of competition and antitrust may not be one other people feel willing or able to engage with that does not resolve the matter.

There are other considerations related to people’s trust choices and trusting supply chains. The document, and reviewers of issues, have categorically stated that people can not trust supply chains. I’ve not seen the evidence for this from any authoritative sources. As a minimum the document should reference such sources for new readers like myself.  Perhaps approaching the problem from the perspective of the conditions under which a person can or can not trust a supply chain would add a much needed dimension to the document?

File: index.bs [hunk 12]
```diff
@@ -949,10 +1041,10 @@ spec:html; type:element; text:link
 <pre class="biblio">
 {
   "COMCAST": {
-      "href": "http://arstechnica.com/tech-policy/2014/09/why-comcasts-javascript-ad-injections-threaten-security-net-neutrality/",
```
Comment (hober) [inline index.bs:952]: Please don't mix substantive textual changes with whitespace changes like these.

Comment (jwrosewell) [reply index.bs]: Reversed in latest commit. Understood. Still learning the W3C document review etiquette.

File: index.bs (hunk 11 above)
Comment (torgo) [reply index.bs:507]: I don't think this is the right forum to have that debate. We're trying to craft some guidelines here, not open up a debate about the nature of the web. Can I suggest that we go with Tess's suggestion and split this out into a new question?

File: index.bs (hunk 8 above)
Comment (torgo) [reply index.bs]: The point that the user agent (vendor) is a party outside of "origins" - that effectively has "root" -  is a good one. Another example is "geolocation" where information is sent out out band to the browser maker as part of the API.

File: index.bs (hunk 11 above)
Comment (jwrosewell) [reply index.bs:507]: Following discussion in the IWA BG where there was broad support to discuss the subject further, and considering the opportunity at TPAC to hold a debate on these matters, I have [proposed a session for October](https://github.com/w3c/web-advertising/issues/87). The output of that session will help inform the future of these definitions and how they apply to the questionairre and the work of the TAG and PING.

It is clear that the current definitions or interpretations of parties is too simplistic to solve the myriad of challanges we are all working on.

Comment (jwrosewell) [reply index.bs:507]: A [TPAC break session proposal](https://github.com/w3c/web-advertising/issues/87#issuecomment-700806304) has been added following the Improving Web Advertising Business Group today and discussion on the session there.

**Comment by hadleybeeman** [comment]
Hi @jwrosewell — we're just checking on this. It looks like there are a bunch of review comments that have come in. Are you still working on these edits?

**Comment by jwrosewell** [comment]
Yes. I've a number of relates issues and changes that I expect to file this week.

**Comment by jwrosewell** [comment]
Following [Texas vs Google](https://www.courtlistener.com/recap/gov.uscourts.txed.202878/gov.uscourts.txed.202878.1.0.pdf) and the material in relation to privacy fixing this PR is taking longer to progress than I'd envisaged.

>"Google actively coordinates with its competitors when it comes to privacy," the suit says. "Of course, effective competition is concerned about both price and quality, and the fact that Google coordinates with its competitors on the quality metric of privacy - one might call it privacy fixing - underscores Google's selective promotion of privacy concerns only when doing so facilitates its efforts to exclude competition".

I expect to return to it early in the New Year.
